src/pages/visualization-view/visualization-view.js

```diff
@@ -30,16 +30,16 @@ const VisualizationView = (props) => {
     isSaveable,
     match: { params: { id: idFromUrl } },
     history,
-    auth: {isAuthenticated}
+    auth: { isAuthenticated }
   } = props
 
   const linkUrl = idFromState && generatePath(routes.visualizationView, { id: idFromState })
   const [localTitle, setLocalTitle] = useState(title || '')
   const startIndex = idFromUrl ? 1 : 0
   const [index, setIndex] = useState(startIndex)
 
-  React.useEffect(() => { reset();  return function cleanup() { reset() }}, [])
-  React.useEffect(() => { if (idFromUrl && idFromUrl !== idFromState) load(idFromUrl);}, [idFromUrl])
+  React.useEffect(() => { reset(); return function cleanup() { reset() } }, [])
+  React.useEffect(() => { if (idFromUrl && !idFromState) load(idFromUrl); }, [idFromUrl, idFromState])
```

Comment:
So if the idFromURL is there AND the idFromState is there, BUT they DON'T match, this will not load from the URL. It will instead will update the URL to match what's in state.  This seems wrong.

Reply:
I'm wondering if this should be simplified to a single useEffect function that looks at both of those fields.  I feel like I don't trust the order in which these are going to execute right now.

Reply:
I know I already said this to you but just for reference: I agree with you but there is a circular check for each of the useEffect bodies, which makes it difficult to put them into one useEffect that is triggered by changes to either `idFromState` or `idFromUrl`